main.js

```diff
@@ -1 +1,145 @@
-  
+ /*
+ * Created by Jenya on 13.06.15.
+ */
+function onScreen (text) {
+    document.getElementById("p1").innerHTML += "<br>" + text;
+}
+
+var textLoaded="";
+var sourceData=[];
+var casesSizes=[];
+var casesTotals=[];
+var casesOutputs=[];
+
+function loadFileAsText() {
+    var fileToLoad = document.getElementById("fileToLoad").files[0];
+    var fileReader = new FileReader();
+    fileReader.onload = function(fileLoadedEvent) {
+        textLoaded = fileLoadedEvent.target.result;
+        divide(textLoaded);
+        loadCases();
+        searcher();
+    };
+    fileReader.readAsText(fileToLoad, "UTF-8");
+}
+
+function loadLocalFile() {
+    var xmlhttp = new XMLHttpRequest();
+    xmlhttp.onreadystatechange = function(){
+        if(xmlhttp.status == 200 && xmlhttp.readyState == 4){
+            textLoaded = xmlhttp.responseText;
+            divide(textLoaded);
+            loadCases();
+            searcher();
+        }
+    };
+    xmlhttp.open("GET","sourse.txt",true);
+    xmlhttp.send();
+}
+
+function divide (rawTaskText) {
```

Comment:
This is another example of function that need a context to run - caseSizes. You better have this function returning a result and be decoupled from globals. Then it is easier to use this function again in other projects